File: Yandex/Practicum/Algorightms_and_Data_Structures/2_sprint/A.Мониторинг/solution.py
import sys

def solution(n: int, m: int, matrix: list[list[str]]) -> None:
    """
    Time Complexity O(NM)
    Memory Complexity O(1)
    """
    for col in range(m):
        for row in range(n):
            if row:
                sys.stdout.write(' ')
            sys.stdout.write(matrix[row][col])
        sys.stdout.write('\n')

if __name__ == '__main__':
    n: int = int(sys.stdin.readline().rstrip())
    m: int = int(sys.stdin.readline().rstrip())
    matrix: list[list[str]] = []
    for _ in range(n):
        row: str = sys.stdin.readline().rstrip().split()
        matrix.append(row)
    solution(n, m, matrix)

File: Yandex/Practicum/Algorightms_and_Data_Structures/2_sprint/A.Мониторинг/solution.cpp
#include <iostream>
#include <vector>

void solution(int n, int m, std::vector<std::vector<int>>& matrix) {
    /*
    Time Complexity O(NM)
    Memory Complexity O(1)
    */
    for (int col = 0; col < m; ++col) {
        for (int row = 0; row < n; ++row) {
            if (row) {
                std::cout << " ";
            }
            std::cout << matrix[row][col];
        }
        std::cout << "\n";
    }
}

int main() {
    int n, m;
    std::cin >> n >> m;
    std::vector<std::vector<int>> matrix;
    for (int r = 0; r < n; ++r) {
        std::vector<int> row;
        for (int c = 0; c < m; ++c) {
            int num;
            std::cin >> num;
            row.push_back(num);
        }
        matrix.push_back(row);
    }
    solution(n, m, matrix);
}